File: QueueImplementation/code.c
#include<stdio.h>
#include<conio.h>
#include<stdlib.h>
#define MAX 10
int queue_arr[MAX];
int rear = -1;
int front = -1;
void insert(int item);
int del();
int peek();
void display();
int isFull();
int isEmpty();

int main(){
	int choice, item;
	while(1){
		printf("\n1. INSERT");
		printf("\n2. DELETE");
		printf("\n3. DISPLAY ELEMENT AT THE FRONT");
		printf("\n4. DISPLAY QUEUE");
		printf("\n5. EXIT");
		printf("\n\nEnter your choice : ");
		scanf("%d",&choice);
		switch(choice){
			case 1: 
				printf("\nEnter the element you want to insert in the queue : ");
				scanf("%d",&item);
				insert(item);
				break;
			case 2:
				item = del();
				printf("\nDequeued element : %d",item);
				break;
			case 3:
				printf("\nElement at the front of the queue : %d",peek());
				break;
			case 4:
				display();
				break;
			case 5:
				exit(1);
				break;
			default:
				printf("\nInvalid choice! Please enter a valid choice.");
		}
	}
}

void insert(int item){
	if(isFull()){
		printf("\nQueue Overflow");
		return;
	}
	if(front == -1)
		front = 0;
	rear++;
	queue_arr[rear] = item;
	return;
}

int del(){
	int item;
	if(isEmpty()){
		printf("\nQueue Underflow");
		exit(1);
	}
	item = queue_arr[front];
	front++;
	return item;
}

int peek(){
	if(isEmpty()){
		printf("\nQueue Underflow");
		exit(1);
	}
	return queue_arr[front];
}

int isEmpty(){
	if(front == -1 || front == rear+1)
		return 1;
	return 0;
}

int isFull(){
	if(rear == (MAX-1))
		return 1;
	return 0;
}

void display(){
	int i;
	if(isEmpty()){
		printf("\nQueue is Empty!!");
		return;
	}
	printf("\nQUEUE : ");
	for(i = front ; i<=rear ; i++)
		printf("%d ",queue_arr[i]);
	printf("\n\n");
}

File: QueueImplementation/code.java
import java.util.Scanner;
class Queue{
    static int MAX = 10;
    int queue_arr[];
    int front, rear;
    Queue(){
        queue_arr = new int[MAX];
        front = -1;
        rear = -1;
    }
    
    boolean isFull(){
        if(front == 0 && rear == MAX-1)
            return true;
        return false;
    }
    
    boolean isEmpty(){
        if(front == -1)
            return true;
        return false;
    }
    
    void enQueue(int item){
        if(isFull()){
            System.out.println("QUEUE IS FULL");
            return;
        }
        else{
            if(front == -1)
                front = 0;
            rear++;
            queue_arr[rear] = item;
        }
        return;
    }
    
    int deQueue(){
        int item;
        if(isEmpty()){
            System.out.println("Queue is empty");
            System.exit(1);
        }
        
        item = queue_arr[front];
        //if the queue has only one element
        if(front>=rear){
            front = -1;
            rear = -1;
        }
        else
            front++;
        return item;
    }
    
    void display(){
        if(isEmpty()){
            System.out.println("Queue is Empty");
            return;
        }
        System.out.println("QUEUE FROM FRONT TO REAR : ");
        for(int i = front ; i <= rear ; i++){
            System.out.print(queue_arr[i]+" ");
        }
        return;
    }
    
    int peek(){
        if(isEmpty()){
            System.out.println("Queue is Empty !");
            System.exit(1);
        }
        return queue_arr[front];
    }
}

public class Main{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		Queue queue = new Queue();
		while(1==1){
		    System.out.println("1. INSERT");
		    System.out.println("2. DELETE");
		    System.out.println("3. DISPLAY ELEMENT AT THE FRONT");
		    System.out.println("4. DISPLAY QUEUE");
		    System.out.println("5. QUIT");
		    System.out.println("Enter the choice : ");
		    int choice = sc.nextInt();
		    switch(choice){
		        case 1:
		            System.out.println("Enter the element you want to insert in the queue : ");
		            int item = sc.nextInt();
		            queue.enQueue(item);
		            break;
		        case 2:
		            item = queue.deQueue();
		            System.out.println("Dequeued element "+item);
		            break;
		        case 3:
		            item = queue.peek();
		            System.out.println("The front of queue : "+item);
		            break;
		        case 4:
		            queue.display();
		            break;
		        case 5:
		            System.exit(1);
		        default:
		            System.out.println("Invalid choice ! Please enter a valid choice .");
		    }
		}
	}
}
